libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/SwitchCondition.cs

```diff
@@ -121,8 +121,27 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
 
                         foreach (var cse in this.Cases)
                         {
-                            // Values for cases are always coerced to string
-                            var caseCondition = Expression.EqualsExpression(this.Condition.ToExpression(), cse.CreateValueExpression());
+                            Expression caseExpression = null;
+
+                            // Values for cases are always coerced to constant
```

Comment:
>// Values for cases are always coerced to constant [](start = 28, length = 50)

Any particular reason you want to force these to be constants?  We don't enforce that for if and there is no particular efficiency concern with giving these the opportunity to be computed.

Reply:
This is just refactoring existing behavior.